app/models/katello/repository.rb

```diff
@@ -483,6 +488,39 @@ def import_system_applicability
       end
     end
 
+    def self.import_distributions
+      self.all.each do |repo|
+        repo.import_distribution_data
+      end
+    end
+
+    def import_distribution_data
+      distribution = Katello.pulp_server.extensions.repository.distributions(self.pulp_id)
+      if distribution.any?
+        distribution = distribution[0]
+        self.update_attributes!(:distribution_version => distribution["version"],
+                                :distribution_arch => distribution["arch"],
+                                :distribution_family => distribution["family"],
+                                :distribution_variant => distribution["variant"],
+                                :distribution_bootable => distribution_bootable?(distribution)
+                                )
+      end
+    end
+
+    def distribution_bootable?(distribution = nil)
```

Comment:
Maybe I structured this wrong. the function distribution_bootable? is there to be called to fill in that attribute in the database. But it also can be called on a repository object to check if it is bootable. I put the distribution argument (line 510) there so it could pass the information that was already called from pulp (line 498), to avoid calling pulp twice in that particular use of the function. Otherwise it calls pulp for the distribution.